ProteoformSuiteGUI/TopDown.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using ProteoformSuiteInternal;
+using System.IO;
+
+namespace ProteoformSuiteGUI
+{
+    public partial class TopDown : Form
+    {
+        private static Color[] colors = new Color[20];
+        private static List<string> mods = new List<string>();
+
+        public TopDown()
+        {
+            InitializeComponent();
+        }
+
+        public void load_dgv()
+        {
+            DisplayUtility.FillDataGridView(dgv_TD_proteoforms, Lollipop.proteoform_community.topdown_proteoforms.Where(p => !p.targeted).ToList());
+            load_colors();
+            load_ptm_colors();
+        }
+
+        public void load_topdown()
+        {
+            if (ready_for_top_down())
+            {
+                run_the_gamut();
+            }
+            else
+            {
+                MessageBox.Show("Go back and load in top-down results.");
+            }
+            cmbx_td_or_e_proteoforms.DataSource = new BindingList<string>() { "TopDown Proteoforms", "Experimental Proteoforms" };
+            cmbx_td_or_e_proteoforms.SelectedIndex = 0;
+        }
+
+        private bool ready_for_top_down()
+        {
+            return (Lollipop.top_down_hits.Count == 0 && Lollipop.input_files.Any(f => f.purpose == Purpose.TopDown));
+        }
+
+        private void run_the_gamut()
+        {
+            if (Lollipop.top_down_hits.Count == 0) read_in_topdown();
+            if (Lollipop.top_down_hits.Count > 0)
+            {
+                Lollipop.proteoform_community.topdown_proteoforms = new TopDownProteoform[0];
+                clear_lists();
+                Lollipop.aggregate_td_hits();
+                if (Lollipop.proteoform_community.topdown_proteoforms.Where(p => p.targeted).Count() > 0) bt_targeted_td_relations.Enabled = true;
+                tb_tdProteoforms.Text = Lollipop.proteoform_community.topdown_proteoforms.Where(p => !p.targeted).ToArray().Length.ToString();
+                load_dgv();
+                bt_td_relations.Enabled = true;
+            }
+        }
+
+        private void read_in_topdown()
+        {
+            if (ready_for_top_down())
+            {
+                 Lollipop.read_in_td_hits();
+            }
+            else
+            {
+                MessageBox.Show("Go back to Load Results and load in top-down results.");
+                return;
+            }
+        }
+
+        private void bt_load_td_Click(object sender, EventArgs e)
+        {
+            run_the_gamut();
+        }
+
+        private void clear_lists()
+        {
+            Lollipop.td_relations.Clear();
+            foreach (Proteoform p in Lollipop.proteoform_community.experimental_proteoforms) p.relationships.RemoveAll(r => r.relation_type == ProteoformComparison.etd);
```

Comment:
That's me trying experimental-targeted-topdown to see what was validated... it's still a work in progress. Right now td proteoforms have a bool to mark if they're targeted. I think it would make more sense to make those ETD relations and then not include them as nodes/edges in the family but then I'm able to see them in the TD form page